addons/sms/models/mail_message.py

```diff
@@ -33,17 +33,16 @@ def _search_has_sms_error(self, operator, operand):
             return ['&', ('notification_ids.notification_status', '=', 'exception'), ('notification_ids.notification_type', '=', 'sms')]
         raise NotImplementedError()
 
-    def _message_format(self, format_reply=True, msg_vals=None, for_current_user=False):
+    def _message_format(self, *args, **kwargs):
         """ Override in order to retrieves data about SMS (recipient name and
             SMS status)
 
         TDE FIXME: clean the overall message_format thingy
         """
-        message_values = super()._message_format(format_reply=format_reply, msg_vals=msg_vals, for_current_user=for_current_user)
-        all_sms_notifications = self.env['mail.notification'].sudo().search([
-            ('mail_message_id', 'in', [r['id'] for r in message_values]),
-            ('notification_type', '=', 'sms')
-        ])
```

Comment:
Note to reviewer:

`self.notification_ids` was already in cache from parent method, this just made an extra query for no reason.

The query didn't scale with number of calls of `_message_format` either, which is necessary in this PR.